docs/introduction.rst

```diff
@@ -107,17 +109,49 @@ structure could be like a simplified XML DOM-tree with following specification:
   (e.g. the unit of the data in the node, etc.). To make things simple, the
   attributes should be text attributes, like in XML.
 
-The sender would assemble a tree with the necessary information and transmit it via
-the transport layer. The receiver would then query the received tree, look
+The sender would assemble a tree with the necessary information and transmit it
+via the transport layer. The receiver would then query the received tree, look
 for the presence / absence of given nodes and extract the necessary information
 from the tree.
 
 I have already started a small C-library with this functionality, the `saydx
-library <https://github.com/saydx/saydx>`_. Although not finished yet, it could
-be used as the message layer. It would provide the basic infrastructure for tree
-manipulation, as well as routines to read and write trees to file or to pass
-them from C to Fortran and vice-versa. Combined with cslib, it could cover the
-functionality of the first two layers.
+[səeɪdɪks] library <https://github.com/saydx/libsaydx>`_. Although not finished
+yet, it could be used as the message layer. It would provide the basic
+infrastructure for tree manipulation, as well as routines to read and write
+trees to file or to pass them from C to Fortran and vice-versa. Combined with
+cslib, it could cover the functionality of the first two layers.
+
+
+Array types
+^^^^^^^^^^^
+
+The message layer should understand / support following array data types:
+
+* real numbers (4, 8 and eventually 16 bytes)
+* complex numbers (composed of two real numbers)
+* integers (4 and 8 bytes)
+* logicals (represented by integers)
+* characters (1 byte)
+* strings (of arbitrary length)
+* bytes
+
+When the data is represented in binary form, the native representation of the
+x86_64 architecture should be used. So, at least in its first version, the
```

Comment:
What about non x86_64? The endian issue is nearly dead, but AIX, Solaris and (pre-2018) NEC are the other way around.

Reply:
I think, we should not cover that for the moment as this will start to become complicated. Especially, what happens if that architecture uses something different as the IEEE-representation. Then swapping bytes won't be enough either.

 We should introduce protocol versions and later protocols could eventually handle it. For the moment, one should resort to sending the data as text, in case the architectures are not compatible.

Reply:
OK, so a compile time flag for defaulting to only text transmission probably should be included.